packages/block-editor/src/components/inserter/menu.js

```diff
@@ -46,6 +47,7 @@ function InserterMenu( {
 		clientId,
 		isAppender,
 		selectBlockOnInsert: __experimentalSelectBlockOnInsert,
+		insertionIndex: __experimentalInsertionIndex,
```

Comment:
I'm concerned about the proliferation of inserter props that all have the same intent, chose where to insert block: `isAppender`, `__experimentalSelectBlockOnInsert`, `__experimentalInsertionIndex`, `rootClientId` and `clientId` are all related to that. I wonder if it's time to take a step back here and figure out something consistent.

Reply:
Yeah, I get that it's adding more along the same lines, especially `clientId` and `__experimentalInsertionIndex`, which are very similarly used.

We could continue making the insertion index more deterministic (more logic inside of `useInsertionPoint`). An option there could be to consider that the root block list in widgets it essentially locked, so maybe `useInsertionPoint` could detect that and instead insert inside the selected block when possible. I'm not sure how that would feel in terms of user experience.

Or alternatively continue on the path of making it configurable, but in a nicer way, make the entire `useInsertionPoint` hook replaceable?

Reply:
> Or alternatively continue on the path of making it configurable, but in a nicer way, make the entire useInsertionPoint hook replaceable?

Yes, something like that might make sense.

I also wonder whether we should remove the responsibility of choosing the position from the inserter component and just have a prop like `{ rootClientId + index }`.

Reply:
That sounds good. I'll do some exploration in a separate PR (this one just merges into #25727).